cpp-client-reference/CMakeLists.txt

```diff
@@ -6,17 +6,41 @@ set(CMAKE_TOOLCHAIN_FILE
 
 project(volition)
 
+include(CheckPIESupported)
+check_pie_supported()
+
 find_package(gRPC CONFIG REQUIRED)
 find_package(Protobuf REQUIRED)
 
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+add_library(common_config INTERFACE)
+target_compile_definitions(common_config INTERFACE
+    $<$<CXX_COMPILER_ID:Clang,GNU>:
+    _FORTIFY_SOURCE=3
+    _GLIBCXX_ASSERTIONS>)
+target_compile_features(common_config INTERFACE cxx_std_20)
+target_compile_options(common_config INTERFACE
+    $<$<CXX_COMPILER_ID:Clang,GNU>:
+    -O2
+    -Wall -Wconversion -Wformat -Wformat=2 -Wimplicit-fallthrough
+    -fno-rtti
+    -fsanitize=address,undefined
+    -fstack-clash-protection
+    #-fstrict-flex-arrays=3
+    -fstack-protector-strong>)
```

Comment:
oh this is the closing `>` for the first `<$`?

I clearly need to spend more time with cmake.

@vchlin dont spend too much time answering my questions here, please just post a good youtube tutorial on cmake 🙃

Reply:
Don't know a good YouTube video offhand. But you'll probably find some if you search for "Modern CMake".